internal/node/controlapi.go

```diff
@@ -69,40 +92,52 @@ func (api *ApiListener) PublicXKey() string {
 }
 
 func (api *ApiListener) Start() error {
-	_, err := api.node.nc.Subscribe(controlapi.APIPrefix+".WPING.>", api.handleWorkloadPing)
-	if err != nil {
-		api.log.Error("Failed to subscribe to workload ping subject", slog.Any("err", err), slog.String("id", api.PublicKey()))
-	}
-	_, err = api.node.nc.Subscribe(controlapi.APIPrefix+".PING", api.handlePing)
+	var sub *nats.Subscription
+	var err error
+
+	sub, err = api.node.nc.Subscribe(controlapi.APIPrefix+".PING", api.handlePing)
 	if err != nil {
 		api.log.Error("Failed to subscribe to ping subject", slog.Any("err", err), slog.String("id", api.PublicKey()))
 	}
+	api.subz = append(api.subz, sub)
```

Comment:
while there's nothing wrong with us manually maintaining a list of subscriptions so that we can un-sub from them later... that's what the `Drain` command on the connection does. All we have to do is drain that once and all the subscriptions that came from that client are disposed of, and likely far faster than doing it 1-by-1.

Reply:
There is an order-of-operations issue with regard to shutting down the machine manager and draining the NATS connections. I think we can remove this manual management of API subscriptions if we change order of operations in `shutdown()`. Will look into this.